### PR TITLE
.NET Standard 2.0 Support

### LINKED ISSUE #15: Create a .NET Standard 2.0 NuGet package
### Description

There is no .NET Core 2.0 version of the NuGet package.

### Repro steps

Build a .NET Core 2.0 project that references FSharp.Collections.ParallelSeq.

### Expected behavior

No warnings.

### Actual behavior

A build warning is logged:

    warning NU1701: Package 'FSharp.Collections.ParallelSeq 1.0.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.

### Related information 

* Windows 7, Linux
* NuGet package 1.0.2
* .NET Core 2.0

### PR DESCRIPTION
**Pull request by ChrSteinert**
Starting to migrate for .NET Standard 2.0.

- [x] Migrate to new SDK style fsproj
- [x] Remove old portable project
- [x] Migrate tests
- [x] Repair build

With the SDK style project file a lot of stuff was dropped. Let me know if something severe is missing.
This will fix #15 !

### CODE REVIEW

**Comment by ChrSteinert** [comment]
Awesome - builds pass.
Ready for reviews! 😄

File: tests/FSharp.Collections.ParallelSeq.Tests/packages.config [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="NUnit" version="2.6.3" targetFramework="net40" />
+  <package id="NUnit" targetFramework="netcoreapp2.0" />
```
Comment (ForNeVeR) [inline tests/FSharp.Collections.ParallelSeq.Tests/packages.config]: Wait, shouldn't we delete the `packages.config` while migrating to the new SDK?

**Comment by ChrSteinert** [comment]
Yes we should. Good catch.

________________________________

**Comment by ChrSteinert** [comment]
If @jamessdixon or @kapilash take a look at this? That would be cool!

File: build.fsx [hunk 2]
```diff
@@ -87,22 +87,16 @@ Target "CleanDocs" (fun _ ->
 // Build library & test project
 
 Target "Build" (fun _ ->
-    !! (solutionFile + "*.sln")
-    |> MSBuildRelease "" "Rebuild"
+    //!! (solutionFile + "*.sln")
```
Comment (kapilash) [inline build.fsx]: Nit: we can remove this commented line as well as the other at line 151

Comment (ChrSteinert) [reply build.fsx]: Sure…

File: src/FSharp.Collections.ParallelSeq/FSharp.Collections.ParallelSeq.fsproj [hunk 3]
```diff
@@ -1,30 +1,8 @@
...
-    <DebugType>full</DebugType>
-    <Optimize>false</Optimize>
-    <Tailcalls>false</Tailcalls>
-    <OutputPath>bin\Debug\</OutputPath>
-    <DefineConstants>DEBUG;TRACE</DefineConstants>
-    <WarningLevel>3</WarningLevel>
-    <DocumentationFile>..\..\bin\FSharp.Collections.ParallelSeq.xml</DocumentationFile>
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (dsyme) [inline src/FSharp.Collections.ParallelSeq/FSharp.Collections.ParallelSeq.fsproj]: As mentioned on twitter it probably makes sense to multi-target here, to ``netstandard2.0;net45``, to allow existing .NET Framework users to continue to use https://twitter.com/atlerudshaug/status/963481134926651392

File: tests/FSharp.Collections.ParallelSeq.Tests/FSharp.Collections.ParallelSeq.Tests.fsproj [hunk 4]
```diff
@@ -1,91 +1,21 @@
...
-    <WarningLevel>3</WarningLevel>
-    <DocumentationFile>bin\Debug\fsharp_project_scaffold_tests.XML</DocumentationFile>
-    <StartAction>Project</StartAction>
-    <StartProgram>
-    </StartProgram>
-    <StartArguments>
-    </StartArguments>
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (dsyme) [inline tests/FSharp.Collections.ParallelSeq.Tests/FSharp.Collections.ParallelSeq.Tests.fsproj]: Do you think we should also test on .NET Framework?

**Comment by ChrSteinert** [comment]
Does that make sense? What are the down sides of consuming a NET Standard 2.0 lib in a NET45 project? Does that not work out of the box? If not, I'd add the multi target 😃

**Comment by Eleanor68** [comment]
A library built with [`.NET Standard 2.0`](https://docs.microsoft.com/en-us/dotnet/standard/net-standard) can be consumed only from `.NET Core 2.0`/`.NET Framework 4.6.1`/`Mono 5.4` projects. Unfortunately, you can not use `.NET Standard 1.0` as target framework because the [`FSharp.Core`](https://www.nuget.org/packages/FSharp.Core/4.3.3) supports only for `.NET Standard 1.6` and `.NET Framework 4.5`.

**Comment by ChrSteinert** [comment]
I see - so net45 is usefull as well. Will make the needed amendments.

**Pull request merged**